sqlmesh/core/macros.py

```diff
@@ -705,28 +723,21 @@ def star(
         raise SQLMeshError(f"Invalid suffix '{suffix}'. Expected a literal.")
     if not isinstance(quote_identifiers, exp.Boolean):
         raise SQLMeshError(f"Invalid quote_identifiers '{quote_identifiers}'. Expected a boolean.")
-    projections: t.List[exp.Alias] = []
-    exclude = set()
-    kwargs = {"quoted": quote_identifiers.this}
-    if alias:
-        kwargs["table"] = alias.name
-    if except_:
-        exclude |= {
-            e.name for e in except_.expressions if isinstance(e, (exp.Identifier, exp.Column))
-        }
-    for column, type_ in evaluator.columns_to_types(relation.sql()).items():
-        if column in exclude:
-            continue
-        projections.append(
-            exp.cast(exp.column(column, **kwargs), type_).as_(
-                f"{prefix.this}{column}{suffix.this}", quoted=kwargs["quoted"]
-            )
+
+    exclude = {e.name for e in except_.expressions}
+    quoted = quote_identifiers.this
+
+    return [
+        exp.cast(exp.column(column, table=alias.name, quoted=quoted), type_).as_(
+            f"{prefix.this}{column}{suffix.this}", quoted=quoted
         )
-    return projections
+        for column, type_ in evaluator.columns_to_types(relation).items()
```

Comment:
I think we need to convert `relation` to a string here. Not sure if `.sql()` suffices because it might generate quotes and the lookup inside the evaluator will probably fail (?).

Reply:
we shouldn't need to,  columns_to_types expects a table

Reply:
Ah I think the type hint is probably wrong then?

```
    # `model_name` should be `str | Table`
    def columns_to_types(self, model_name: str) -> t.Dict[str, exp.DataType]:
```

Reply:
@tobymao `relation` isn't actually a `Table` though, right? I believe it will be parsed into a `Column` by the macro evaluator and so this will fail inside of `exp.to_table` (which is invoked in `MacroEvaluator.columns_to_Types`)

Reply:
i handle that

Reply:
Ah sorry I just saw the update 👍